src/components/color-picker/color-picker.tsx

```diff
@@ -1087,21 +1090,25 @@ export class ColorPicker implements InteractiveComponent {
 
   async fetchStrings(): Promise<void> {
     // TODO: these could split into reusable utils
+    // setup util for fetching locale
+
+    //fetch supported locales
     const locale = this.getLocale();
     // TODO: fetching could be moved to a util (similar to icon fetching)
-    this.builtInStrings = await (
-      await fetch(getAssetPath(`./assets/color-picker/i18n/${locale}.json`))
-    ).json();
+    this.builtInStrings = await fetchLocaleStrings(locale, "color-picker");
   }
 
   getLocale(): string {
-    const lang = document.documentElement.lang || navigator.language; // need to determine best place to grab locale from
+    // need to determine best place to grab locale from
+    //we can go by order ( locale of component > locale of closests > lang of document > navigator)
+    const lang = document.documentElement.lang || navigator.language;
```

Comment:
Let's include the element's `lang` with higher precedence:

`el.lang || document.documentElement.lang || navigator.language;`

Reply:
It should be `el.locale` which gets the preference.

Reply:
atleast for the components with `locale` prop. We may need a util method that determines locale for a given component considering the element's locale, lang, closest elements locale & document.documentElement.lang || navigator.language. Thoughts?

Reply:
We're going to be dropping `locale` in favor of the global `lang` attribute/prop. 

I agree that we need to think about looking up `lang` if set on a parent vs the component itself, but that's outside the scope of this PR. I'll create an issue for this.

Reply:
Lookup and resolution of locale is handled by https://github.com/Esri/calcite-components/pull/5338